server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v66/CreateTableAnalysisProperties.java

```diff
@@ -0,0 +1,91 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.platform.db.migration.version.v66;
+
+import java.sql.SQLException;
+import org.sonar.db.Database;
+import org.sonar.server.platform.db.migration.def.BigIntegerColumnDef;
+import org.sonar.server.platform.db.migration.def.BooleanColumnDef;
+import org.sonar.server.platform.db.migration.def.ClobColumnDef;
+import org.sonar.server.platform.db.migration.def.VarcharColumnDef;
+import org.sonar.server.platform.db.migration.sql.CreateIndexBuilder;
+import org.sonar.server.platform.db.migration.sql.CreateTableBuilder;
+import org.sonar.server.platform.db.migration.step.DdlChange;
+
+public class CreateTableAnalysisProperties extends DdlChange {
+  private static final String TABLE_NAME = "analysis_properties";
+  private static final String SNAPSHOT_UUID = "snapshot_uuid";
+
+  public CreateTableAnalysisProperties(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    context.execute(new CreateTableBuilder(getDialect(), TABLE_NAME)
+      .addPkColumn(VarcharColumnDef.newVarcharColumnDefBuilder()
+        .setColumnName("uuid")
+        .setIsNullable(false)
+        .setLimit(40)
+        .build())
+      .addColumn(VarcharColumnDef.newVarcharColumnDefBuilder()
+        .setColumnName(SNAPSHOT_UUID)
+        .setIsNullable(false)
+        .setLimit(40)
+        .build())
+      .addColumn(VarcharColumnDef.newVarcharColumnDefBuilder()
+        .setColumnName("kee")
+        .setIsNullable(false)
+        .setLimit(512)
+        .build())
+      .addColumn(VarcharColumnDef.newVarcharColumnDefBuilder()
+        .setColumnName("text_value")
+        .setIsNullable(true)
+        .setLimit(4000)
+        .build())
+      .addColumn(ClobColumnDef.newClobColumnDefBuilder()
+        .setColumnName("clob_value")
+        .setIsNullable(true)
+        .build())
+      .addColumn(BooleanColumnDef.newBooleanColumnDefBuilder()
+        .setColumnName("is_empty")
+        .setIsNullable(false)
+        .build())
+      .addColumn(BigIntegerColumnDef.newBigIntegerColumnDefBuilder()
+        .setColumnName("created_at")
+        .setIsNullable(false)
+        .build())
+      .build()
+    );
+
+    context.execute(new CreateIndexBuilder(getDialect())
+      .addColumn(VarcharColumnDef.newVarcharColumnDefBuilder()
+        .setColumnName(SNAPSHOT_UUID)
+        .setIsNullable(false)
+        .setLimit(40)
+        .build())
+      .setUnique(false)
+      .setTable(TABLE_NAME)
+      .setName(SNAPSHOT_UUID)
```

Comment:
I don't think it's wise to use the column name as the index name. (in fact, we don't do that)

usual patterns are "IX_{column_name}" or "{table_name}_{column_name}" or "IX_{table_name}_{column_name}" (mind the maximum length, but the builder should fail if too long)

Reply:
Indeed I have a look at schema-h2.ddl but I was not able to determine which pattern should be used. And I found many with the column name but I was wrong. Good to know.